base/compiler/typelimits.jl

```diff
@@ -414,8 +414,12 @@ function tmerge(@nospecialize(typea), @nospecialize(typeb))
             fields = Vector{Any}(undef, type_nfields)
             anyconst = false
             for i = 1:type_nfields
-                ity = tmerge(getfield_tfunc(typea, Const(i)),
-                             getfield_tfunc(typeb, Const(i)))
+                ai = getfield_tfunc(typea, Const(i))
+                bi = getfield_tfunc(typeb, Const(i))
+                ity = tmerge(ai, bi)
+                if ai === Union{} || bi === Union{}
+                    ity = widenconst(ity)
+                end
```

Comment:
```suggestion
                if ai === Union{} || bi === Union{}
                    ai = widenconst(ai)
                    bi = widenconst(bi)
                end
                ity = tmerge(ai, bi)
```

Reply:
Sure, that should work as well, but why would it be better? (If one of them is `Union{}`, `tmerge` will return the other, whether its `widenconst`ed or not, so the result should be the same, no?)

Reply:
true, that would likely not make a difference in this case